### DISCUSSION TITLE
Context to spark submit operator

### DISCUSSION THREAD

**Discussion by sou-joshi** [apache/airflow#23295] [Q&A]
Is there a way we can pass the context to the Spark submit operator?
I have tried passing few variables required as args and works fine. But i need the information of all the tasks to be passed to a spark job. Is there a way to do this?

**Comment by potiuk** [comment]
You should use JINJA. https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html . If you need whole context and jinja is not enough - create your operator deriving from Spark SubmitOperator and override it's "execute" method. Airflow DAGs are Python and you can simply do whatever Python allows you. Extending an operator is typical thing to do.
